src/FlowChart.hs

```diff
@@ -0,0 +1,244 @@
+module FlowChart where
+
+import Algebra.Graph.Labelled as LG
+import Relude.Extra.Foldable1
+import System.Directory.Internal.Prelude ()
+import Text.Megaparsec hiding (State)
+import Text.Megaparsec as M
+import Text.Megaparsec.Char (alphaNumChar, char, space1, string)
+import qualified Text.Megaparsec.Char.Lexer as L
+
+type Parser = Parsec Void Text
+
+data Diagram
+  = FlowChart
+      { orientation :: Orientation,
+        graph :: FlowChartGraph
+      }
+  | Others
+  deriving (Eq, Show)
+
+data Orientation = TB | TD | BT | RL | LR deriving (Eq, Show)
+
+type FlowChartGraph = Graph Edge [Node]
+
+data Edge = Edge
+  { edgeStyle :: Text,
+    edgeLabel :: Maybe Text
+  }
+  deriving (Eq, Show)
+
+instance Semigroup Edge where
+  (<>) (Edge a b) (Edge c d) = Edge (a <> c) (b <> d)
+
+instance Monoid Edge where
+  mempty = Edge "" Nothing
+
+data Node = Node
+  { nodeId :: Text,
+    nodeLabel :: Maybe (Bracket, Text)
+  }
+  deriving (Eq, Show, Ord)
```

Comment:
This is a bit confusing. Why is `Bracket` part of "label"? Aren't they different things? Try the following:

```haskell
data Node = Node 
  { nodeId :: NodeId,
    nodeLabel :: Maybe NodeLabel,
    nodeShape :: Maybe Shape
  }

data Shape = Rhombus | Round | ...

newtype NodeId = NodeId { unNodeId :: Text } 
```

- Reading the type should more or less tell us the semantics of the data being represented in a straightforward way.
- We use `newtype` wrapper over primitive types. Seeing `Text` peppered throughout in *different* semantic context can be confusing (for eg., when reading the parser code, all of which return `Parser Text` without indicating the type of text being returned).